plugins/modules/site_workflow_manager.py

```diff
@@ -768,6 +1137,206 @@ def get_want(self, config):
+                op_modifies=True,
+                params=floor_param
+            )
+
+            if response and isinstance(response, dict):
+                self.log("Initial API response from 'updates_a_floor': {0}".format(
+                    response), "DEBUG")
+                executionid = response.get("executionId")
+                if executionid:
+                    while True:
+                        execution_details = self.get_execution_details(
+                            executionid)
+                        if execution_details.get("status") == "SUCCESS":
+                            self.log(
+                                "Floor update completed successfully.", "INFO")
+                            break
+                        elif execution_details.get("bapiError"):
+                            self.log("Error during floor update execution: {0}".format(
+                                execution_details.get("bapiError")), "ERROR")
+                            self.module.fail_json(msg=execution_details.get(
+                                "bapiError"), response=execution_details)
+                            break
+            return response
+
+        except Exception as e:
+            error_msg = "Exception occurred while updating floor with site_id '{0}' due to: {1}".format(
+                site_params.get('site_id'), str(e)
+            )
+            self.log(error_msg, "ERROR")
+        return self
+
+    def update_building(self, site_params):
+        """
+        Updates a building in the site hierarchy using the provided site id.
+
+        Args:
+            site_params (dict): Dictionary containing parameters required for the building update, including the site_id.
+
+        Returns:
+            dict: The API response from the 'update_a_building' operation.
+        """
+        try:
+            self.log("Updating building with parameters: {0}".format(
+                site_params), "INFO")
+            parent_id = self.have.get("parent_id")
+            site_params['site']['building']['parentId'] = parent_id
+            self.log("Updated site_params with parent_id: {0}".format(
+                site_params), "INFO")
+            building_param = site_params.get('site', {}).get('building')
+            site_id = self.have.get("site_id")
+            building_param['id'] = site_id
+
+            response = self.dnac._exec(
+                family="site_design",
+                function='updates_a_building',
+                op_modifies=True,
+                params=building_param,
+            )
+
+            self.log("Building update successful. API response: {0}".format(
+                response), "DEBUG")
+            return response
+
+        except Exception as e:
+            error_msg = "Exception occurred while updating building '{0}' due to: {1}".format(
+                site_params.get('site_name'), str(e)
+            )
+            self.log(error_msg, "ERROR")
+        return self
```

Comment:
In each update function you are returning self and response how you handle this ?

Reply:
response when the building update is successful, and return self on failure(exception)

Reply:
That is not allowed. Either return response or self.